packages/blade/src/components/DatePicker/DateInput.web.tsx

```diff
@@ -0,0 +1,186 @@
+  const hasLabel = Boolean(label);
+  const { locale } = useDatesContext();
+
+  if (selectionType == 'single') {
+    return (
+      <DateInput
+        ref={ref as never}
+        id="start-date"
+        labelPosition={labelPosition}
+        label={label}
+        placeholder={format}
+        popupId={referenceProps['aria-controls']}
+        isPopupExpanded={referenceProps['aria-expanded']}
+        size={size}
+        value={getFormattedDate({
+          date,
+          format,
+          labelSeparator: '-',
+          locale,
+          type: 'default',
+        })}
+        {...props}
+        {...referenceProps}
+      />
+    );
+  }
+
+  if (selectionType == 'range') {
+    return (
+      <BaseBox
+        width="100%"
+        display="flex"
+        flexDirection="row"
+        gap="spacing.4"
+        alignItems="flex-end"
+        ref={ref as never}
+      >
+        <BaseBox flex={1}>
+          <DateInput
+            setInputWrapperRef={(node) => ((inputRef as any)!.current = node)}
+            id="start-date"
+            leadingIcon={CalendarIcon}
+            label={label?.start}
+            labelPosition={labelPosition}
+            placeholder={format}
+            popupId={referenceProps['aria-controls']}
+            isPopupExpanded={referenceProps['aria-expanded']}
+            size={size}
+            value={getFormattedDate({
+              type: 'default',
+              date: date[0],
+              format,
+              labelSeparator: '-',
+              locale,
+            })}
+            {...props}
+            {...referenceProps}
+          />
+        </BaseBox>
+        <BaseBox flexShrink={0} alignSelf="start">
+          <ArrowRightIcon
+            size="medium"
+            marginTop={
+              // Hacky layouting because the we cannot put this inside the internal layout of BaseInput.
+              hasLabel && !isLabelPositionLeft
+                ? `calc(${makeSize(iconVerticalMargin[size])} + ${makeSize(
+                    isLarge ? sizeTokens[20] : sizeTokens[15],
+                  )})`
+                : makeSize(iconVerticalMargin[size])
+            }
+          />
+        </BaseBox>
+        <BaseBox flex={1}>
+          <DateInput
+            id="end-date"
+            placeholder={format}
+            leadingIcon={CalendarIcon}
+            label={isLabelPositionLeft ? undefined : label?.end}
+            labelPosition={isLabelPositionLeft ? undefined : labelPosition}
+            popupId={referenceProps['aria-controls']}
+            isPopupExpanded={referenceProps['aria-expanded']}
+            size={size}
+            value={getFormattedDate({
+              type: 'default',
+              date: date[1],
+              format,
+              labelSeparator: '-',
+              locale,
+            })}
+            {...props}
+            {...referenceProps}
+          />
+        </BaseBox>
+      </BaseBox>
+    );
+  }
+
+  return <></>;
```

Comment:
Just have if & else? Imagine someone actually passes invalid JS and you actually see an empty date input

Reply:
React components needs to return something. Otherwise TS will throw error. 

This case will never actually suffice because selectionType will always be set to `single` by default. So no way input can be hidden.